src/app/App.js

```diff
@@ -651,6 +657,20 @@ class App extends EventEmitter {
 		return path.indexOf(this.basePath) === 0;
 	}
 
+	/**
+	 * Tests if the given path is the same of the current pendingNavigate.
+	 * @param {!string} path Link path containing the querystring part.
+	 * @return {boolean}
+	 * @protected
+	 */
+	isSamePendingNavigationPath_(path) {
+		if (this.isNavigationPending && utils.getUrlPath(this.pendingNavigate.path) === path) {
```

Comment:
When just using `this.pendingNavigate.path` to compare with the given `path` it won't work because `this.pendingNavigate.path` is in a URL format and not in a path(starting with `/`).